integration_tests/route_helpers.go

```diff
@@ -97,3 +105,36 @@ func clearRoutes() {
 	routerDB.C("routes").DropCollection()
 	routerDB.C("backends").DropCollection()
 }
+
+func getMongoSession(uri string) *mgo.Session {
```

Comment:
Could we extract this out into its own package to reduce duplication between here and `router.go`? That would also allow us to unit test it. I think this would probably be a bit easier to follow as a few shorter methods.